cmd/argo/commands/resume.go

```diff
@@ -18,15 +29,23 @@ func NewResumeCommand() *cobra.Command {
 				cmd.HelpFunc()(cmd, args)
 				os.Exit(1)
 			}
+			parsedStepSelector, err := fields.ParseSelector(resumeArgs.stepSelector)
+			if err != nil {
+				log.Fatal(err)
+			}
+
 			InitWorkflowClient()
 			for _, wfName := range args {
-				err := util.ResumeWorkflow(wfClient, wfName)
+				err := util.ResumeWorkflow(wfClient, wfName, resumeArgs.stepFail, resumeArgs.stepFailMessage, parsedStepSelector)
 				if err != nil {
 					log.Fatalf("Failed to resume %s: %+v", wfName, err)
 				}
 				fmt.Printf("workflow %s resumed\n", wfName)
 			}
 		},
 	}
+	command.Flags().BoolVar(&resumeArgs.stepFail, "step-fail", false, "Fail suspended step")
```

Comment:
let me get this right, instead of resuming the workflow, we actually fail the suspend step with a message?

Reply:
That's correct.

Reply:
This is a very different operation to resuming the workflow, you are not in fact resuming it at all - you are failing it.

Am I right? If so, we should probably add an `argo fail` command.

Reply:
That's a reasonable approach - I can do that when I get further time to work on this PR.

Reply:
@jessesuen @simster7 thoughts?

Reply:
@alexec @jessesuen @simster7 

Just checking where we're at here so I can get this PR ready for merge.

Reply:
So sorry for the delay... I agree with @alexec. `argo fail` would be better for this

Reply:
@simster7 @alexec 
Before I move forward, a quick idea - could it be clearer to rename argo resume/fail to argo unsuspend ? That would make it clearer that the command is only relevant in the context of a suspended workflow.

Reply:
@mark9white I've been thinking... how about we keep this PR with `argo fail <SELECTOR>` and also make #2352 work under `argo fail` instead of `argo stop`. Abstractly, both this PR and #2352 fail the node/workflow. What do you think?

Reply:
Sure, that makes sense to me.

Reply:
Ok great. I will make the changes to #2352 and after that's merged you'll be able to use the command infrastructure to add this.

Reply:
@simster7 given the change of approach in #2352 are you looking to have this PR implement something like `argo stop --fail <selector>`?

Reply:
@simster7 a little nudge ;)

Reply:
Sorry about that. It would be something like `argo stop --nodeSelector`. `--fail` is the default and is implied at this point.

Reply:
We want to distinguish between stoppin a particular node and the workflow as a whole